### PR TITLE
[FEATURE] Document the ModuleTemplateFactory

### PR DESCRIPTION
**Pull request by linawolf**
Releases: main, 12.4

### CODE REVIEW

File: Documentation/ApiOverview/Backend/BackendModules/ModuleTemplateFactory.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. include:: /Includes.rst.txt
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/ModuleTemplateFactory.rst:1]: 2 spaces

File: Documentation/ApiOverview/Backend/BackendModules/ModuleTemplateFactory.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+.. index:: Backend modules; ModuleTemplateFactory
+.. _ModuleTemplateFactory:
+
+=====================
+ModuleTemplateFactory
+=====================
+
+The template module factory should be used by backend controllers to create a
+:php:`\TYPO3\CMS\Backend\Template\ModuleTemplate.`
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/ModuleTemplateFactory.rst:10]: Dot should not be part of class

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        $moduleTemplate = $this->initializeModuleTemplate($this->request);
+
+        $moduleTemplate->assignMultiple([
+            'variable1' => 'value 1',
+            'variable2' => 'value 2',
+        ]);
+        return $moduleTemplate->renderResponse('Backend/Some');
+    }
+
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php:37]: Remove empty space

File: Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplateFactory.rst.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+..  Generated by https://github.com/TYPO3-Documentation/t3docs-codesnippets
+..  php:namespace::  TYPO3\CMS\Backend\Template
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplateFactory.rst.txt:2]: After `::` just one space

File: Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplateFactory.rst.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+..  php:class:: ModuleTemplateFactory
+
+    A factory class creating backend related ModuleTemplate view objects.
+
+    ..  php:method:: create(Psr\Http\Message\ServerRequestInterface $request)
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplateFactory.rst.txt:8]: Add `\` in front of class

**Review by froemken — CHANGES_REQUESTED** [review]

**Comment by linawolf** [comment]
Solved with https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/4416

**Pull request closed without merging**